src/mgmt/cli_mgmt.c

```diff
@@ -604,6 +616,13 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 	g_num_serv_ranks = resp->n_rank_uris;
 	D_INFO("Setting number of server ranks to %d\n", g_num_serv_ranks);
 
+	/* Save ranks id */
+	D_ALLOC_ARRAY(g_serv_ranks, g_num_serv_ranks);
```

Comment:
Either this function will need to be prevented from being called multiple times or previous g_serv_ranks allocation needs to be free-ed first here, or re-alloced

Reply:
Yep, I was going to test if it could be called multiple times.  From my side, I will be in favour of realloacating (i.e. refreshing the rank table index).
@mchaarawi and @mchaarawi , I am also not sure if some special care should be taken with the function https://github.com/daos-stack/daos/blob/57701fa5147f96e63f066f712d2947efaf7cb28f/src/client/api/event.c#L101

Reply:
I dont think there is a concurrent access to this function; i think the main case would be as you mentioned earlier -- g_serv_ranks might be already allocated in the forked child, so just need to handle that case properly by handling re-alloc/re-use of this global buffer here

Reply:
> Yep, I was going to test if it could be called multiple times. From my side, I will be in favour of reallocating (i.e. refreshing the rank table index). @frostedcmos , do you know if we could have some concurrency issues with this function.
> 
> * [x]  Reallocate` g_serv_ranks`

Fixed with commit